collectors/osd.go

```diff
@@ -0,0 +1,492 @@
+
+	OsdIn *prometheus.GaugeVec
+
+	OsdUp *prometheus.GaugeVec
+
+	TotalKB prometheus.Gauge
+
+	TotalUsedKB prometheus.Gauge
+
+	TotalAvailKB prometheus.Gauge
+
+	AverageUtil prometheus.Gauge
+}
+
+func NewOsdCollector(conn Conn) *OsdCollector {
+	return &OsdCollector{
+		conn: conn,
+
+		CrushWeight: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_crush_weight",
+				Help:      "OSD Crush Weight",
+			},
+			[]string{"osd"},
+		),
+
+		Depth: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_depth",
+				Help:      "OSD Depth",
+			},
+			[]string{"osd"},
+		),
+
+		Reweight: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_reweight",
+				Help:      "OSD Reweight",
+			},
+			[]string{"osd"},
+		),
+
+		KB: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_kb",
+				Help:      "OSD Total KB",
+			},
+			[]string{"osd"},
+		),
+
+		UsedKB: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_kb_used",
+				Help:      "OSD Used Storage in KB",
+			},
+			[]string{"osd"},
+		),
+
+		AvailKB: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_kb_avail",
+				Help:      "OSD Available Storage in KB",
+			},
+			[]string{"osd"},
+		),
+
+		Utilization: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_utilization",
+				Help:      "OSD Utilization",
+			},
+			[]string{"osd"},
+		),
+
+		Pgs: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_pgs",
+				Help:      "OSD Placement Group Count",
+			},
+			[]string{"osd"},
+		),
+
+		TotalKB: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_total_kb",
+				Help:      "OSD Total Storage KB",
+			},
+		),
+		TotalUsedKB: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_total_kb_used",
+				Help:      "OSD Total Used Storage KB",
+			},
+		),
+
+		TotalAvailKB: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_total_kb_avail",
+				Help:      "OSD Total Available Storage KB ",
+			},
+		),
+
+		AverageUtil: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_average_utilization",
+				Help:      "OSD Average Utilization",
+			},
+		),
+
+		CommitLatency: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_perf_commit_latency_ms",
+				Help:      "OSD Perf Commit Latency",
+			},
+			[]string{"osd"},
+		),
+
+		ApplyLatency: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_perf_apply_latency_ms",
```

Comment:
Same as [this](https://github.com/digitalocean/ceph_exporter/pull/26/files/62705b90b2f6c2d58fac7dc167a2cb41046519d5#r72252289).